website/builder/markdown.py

```
@@ -649,8 +681,43 @@ def render_toc(self, html_content: str) -> str:
+                if m:
+                    display_text = m.group(1).strip()
+                else:
+                    display_text = heading_id
+            display_text = _html.unescape(display_text)
+
+            # If no icon and this is a leaf heading, use the default SVG
+            if not icon_html and not has_child:
+                icon_html = default_svg
+
+            # Add the current heading with icon (if any)
+            toc_html += f'<li><a href="#' + heading_id + f'">{icon_html}{display_text}</a></li>\n'
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)